dynamiclinks/DynamicLinksSwiftUIExample/DynamicLinksExampleApp.swift

```diff
@@ -16,27 +16,58 @@ import SwiftUI
 import FirebaseCore
 import FirebaseDynamicLinks
 
-class AppDelegate: NSObject, UIApplicationDelegate {
-  func application(_ application: UIApplication,
-                   didFinishLaunchingWithOptions launchOptions: [UIApplication
-                     .LaunchOptionsKey: Any]? = nil) -> Bool {
-    FirebaseApp.configure()
-    DynamicLinks.performDiagnostics()
-
-    return true
-  }
-}
-
 @main
 struct DynamicLinksExampleApp: App {
   @UIApplicationDelegateAdaptor(AppDelegate.self) var delegate
   @StateObject var model = LinkModel()
+  @State var alertPresented: Bool = false
+  @State var alertMessage: String = "URL: empty"
+  @State var receivedLinkModel: ReceivedLinkModel?
 
   var body: some Scene {
     WindowGroup {
       NavigationView {
         LinkConfigurationView()
       }.environmentObject(model)
+        .onOpenURL { url in
+          let dynamicLinks = DynamicLinks.dynamicLinks()
+          if dynamicLinks.shouldHandleDynamicLink(fromCustomSchemeURL: url) {
+            print("The URL \(url) has an FDL Custom Scheme; should be handled.")
+            if let dynamicLink = dynamicLinks.dynamicLink(fromCustomSchemeURL: url) {
+              receivedLinkModel = ReceivedLinkModel(receivedURL: url, dynamicLink: dynamicLink)
+            } else {
+              print("The URL \(url) is not an FDL Dynamic Link.")
+            }
+          } else {
+            print(
+              "The URL \(url) does not have an FDL Custom Scheme; should be handled as Universal URL."
+            )
+            Task {
+              if let dynamicLink: DynamicLink = await handleLink(url) {
+                receivedLinkModel = ReceivedLinkModel(receivedURL: url, dynamicLink: dynamicLink)
+              }
+            }
+          }
+        }
+        .sheet(item: $receivedLinkModel, content: { LinkReceivedView(receivedLinkModel: $0) })
+    }.handlesExternalEvents(matching: ["*"])
+  }
+
+  func handleLink(_ url: URL) async -> DynamicLink? {
+    await withCheckedContinuation { continuation in
+      handleLink(url) { dynamicLink, error in
+        // TODO: Handle error
+        continuation.resume(returning: dynamicLink)
+      }
+    }
+  }
+
+  func handleLink(_ url: URL, completion: @escaping (DynamicLink?, Error?) -> Void) {
```

Comment:
Can this be `async throws` instead?

Reply:
Good idea. This code has moved to [`extractDynamicLink`](https://github.com/firebase/quickstart-ios/pull/1428/files#diff-e25e2955aedb4999ed4a8c5d7f7bc26a9b474bfaaf1a0f92dc768e02ecc00100R101) but I made the change there.